packages/web/src/components/track/GiantTrackTile.js

```diff
@@ -303,22 +314,66 @@ class GiantTrackTile extends PureComponent {
     )
   }
 
+  renderProgressInfo() {
```

Comment:
it might be nice to isolate this component so we can also isolate it's selectors for positionInfo etc? i know a lot of these components aren't really set up for this though. another win is that we can write this isolated component in typescript too. lmk and i can help set that up since i have experience adding small ts components in a large web js component